conmon-rs/server/src/pause.rs

```diff
@@ -0,0 +1,266 @@
+use anyhow::{bail, Context, Result};
+use capnp::enum_list::Reader;
+use conmon_common::conmon_capnp::conmon;
+use getset::{CopyGetters, Getters};
+use libc::pid_t;
+use nix::{
+    mount::{mount, umount, MsFlags},
+    sched::{unshare, CloneFlags},
+    sys::signal::{kill, Signal},
+    unistd::{fork, ForkResult, Pid},
+};
+use once_cell::sync::OnceCell;
+use signal_hook::{consts::TERM_SIGNALS, iterator::Signals};
+use std::{
+    env,
+    fs::{self, File},
+    io::Write,
+    path::{Path, PathBuf},
+    process::{exit, Command},
+};
+use strum::{AsRefStr, Display, EnumIter, EnumString, IntoStaticStr};
+use tracing::{debug, error, info, warn};
+use uuid::Uuid;
+
+/// The main structure for this module.
+#[derive(Debug, CopyGetters, Getters)]
+pub struct Pause {
+    #[get = "pub"]
+    path: PathBuf,
+
+    #[get = "pub"]
+    namespaces: Vec<Namespace>,
+
+    #[get_copy]
+    pid: Pid,
+}
+
+/// The global shared multiple pause instance.
+static PAUSE: OnceCell<Pause> = OnceCell::new();
+
+/// The global path for storing bin mounted namespaces.
+const PAUSE_PATH: &str = "/var/run/conmonrs";
+
+/// The file path for storing the pause PID.
+const PAUSE_PID_FILE: &str = ".pause_pid";
```

Comment:
Would it make sense to run this as a systemd unit instead?

Reply:
Hm, I'm not sure about the resource overhead we add when using systemd here. We can definitely give it a try and see how it behaves.